### DISCUSSION TITLE
Page load time and first tiles

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre-gl-js#2282] [Ideas]
I wonder if it would be possible to bring down the page load time of MapLibre GL JS by bundling the tiles needed for the first view together with the javascript library?

Or maybe the tiles would not have to be bundled with the library, but when the client requests the `maplibre-gl.js` package it could at the same time also get some first tiles such that they are in the browser cache already. Do you think that would work?

**Comment by HarelM** [comment]
I think it's possible, but I'm not sure you'll be able to do it as part of maplibre-gl-js (your second idea anyway).
Worth exploring nevertheless.

**Reply by AbelVM** [reply]
I'd love to take a look at that. It sounds quite connected to https://github.com/maplibre/maplibre-gl-js/issues/116
